### PR TITLE
store: allocate vector with capacity

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by ngaut** [comment]
PTAL

File: src/raftstore/store/peer_storage.rs [hunk 1]
```diff
@@ -189,7 +189,7 @@ impl PeerStorage {
 
     pub fn entries(&self, low: u64, high: u64, max_size: u64) -> raft::Result<Vec<Entry>> {
         try!(self.check_range(low, high));
-        let mut ents = vec![];
+        let mut ents = Vec::with_capacity((high - low) as usize);
```
Comment (BusyJay) [inline src/raftstore/store/peer_storage.rs]: This may allocate too much memory when max_size is not NO_LIMIT.

Comment (ngaut) [inline src/raftstore/store/peer_storage.rs]: We will allocate (high - low) entries anyway.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**